### PR TITLE
FPS counter in tutorial game

### PR DESCRIPTION
**Pull request by AdSand**
## Initial setup for the Debug mode.

Uses a debug mode class that contains all the basic functions needed for debug mode.
To use, create a debug mode object in the game, and call the DisplayDebug function, passing the game's dt as a parameter.

Currently only the FPS counter is working in tutorial game

### CODE REVIEW

File: Client/DebugMode.cpp [hunk 1]
```diff
@@ -0,0 +1,30 @@
+#include "DebugMode.h"
+
+void NCL::CSC8503::DebugMode::DisplayDebug(float dt)
+{
+	GetFPSCount(dt);
+	GetMemoryUsage();
+}
+
+void NCL::CSC8503::DebugMode::GetFPSCount(float dt)
```
Comment (DelCherub) [inline Client/DebugMode.cpp]: "Get" suggests that the function will return the FPS count: this should probably be "DisplayFPSCount" instead, likewise for GetMemoryUsage

**Review by DelCherub — COMMENTED** [review]
Looks fine to me apart from changing function names.

File: Client/CMakeLists.txt [hunk 2]
```diff
@@ -1,6 +1,6 @@
 cmake_minimum_required(VERSION 3.16.0 FATAL_ERROR)
 project(client)
 
-add_executable(client main.cpp Client.cpp Renderer.cpp TutorialGame.cpp Font.cpp)
+add_executable(client main.cpp Client.cpp Renderer.cpp TutorialGame.cpp Font.cpp DebugMode.cpp)
```
Comment (St4rman) [inline Client/CMakeLists.txt]: There's a better way to do this, will send a corresponding message/show in lab. This is a fine way though

File: Client/DebugMode.cpp (hunk 1 above)
Comment (St4rman) [reply Client/DebugMode.cpp]: Agreed

File: Client/TutorialGame.cpp [hunk 3]
```diff
@@ -64,6 +66,9 @@ TutorialGame::~TutorialGame()	{
 	delete physics;
 	delete renderer;
 	delete world;
+
+	delete debugMode;
```
Comment (St4rman) [inline Client/TutorialGame.cpp:73]: Chad memory managing. Nice one. :D

**Review by St4rman — APPROVED** [review]
Looks relatively okay to me.  I'm sure that there are more stats we'll eventually need to be able to add but this is a decent starting out point. 

Good to merge AFTER merge conflicts solved and changing the method names. 😄

**Review by Odycious — APPROVED** [review]
I have made the requested changes by Matthew and Ed.

**Pull request merged**